drivers/net/ethernet/intel/idpf/idpf_eth_idc.h

```diff
@@ -0,0 +1,199 @@
+	IDPF_QUEUE_DEFAULT_VPORT = 1,
+	IDPF_QUEUE_DYNAMIC_VPORT
+};
+
+/**
+ * enum idpf_eth_idc_event_type
+ * Event type IDPF_ETH_IDC_EVENT_ALL_VPORTS are routed to all ports, while
+ * the event type IDPF_ETH_IDC_EVENT_SINGLE_VPORT is routed to only the
+ * intended port.
+ */
+enum idpf_eth_idc_event_type {
+	IDPF_ETH_IDC_EVENT_ALL_VPORTS,
+	IDPF_ETH_IDC_EVENT_SINGLE_VPORT
+};
+
+enum idpf_eth_idc_event_code {
+	/* Following events are from main driver to aux/eth driver */
+	IDPF_ETH_IDC_EVENT_LINK_CHANGE,
+	IDPF_ETH_IDC_EVENT_RESET_INITIATED,
+	IDPF_ETH_IDC_EVENT_RESET_COMPLETE,
+	IDPF_ETH_IDC_EVENT_DEINIT,
+	IDPF_ETH_IDC_EVENT_REMOVE,
+	IDPF_ETH_IDC_EVENT_PROBE_POST_INIT,
+
+	/* Following requests are from auxiliary eth driver to main driver */
+	IDPF_ETH_IDC_EVENT_REQ_HARD_RESET,
+	IDPF_ETH_IDC_EVENT_ENTER_PROMISCUOUS_MODE,
+};
+
+/**
+ * struct idpf_eth_idc_auxiliary_dev_caps - Capabilities info
+ * @csum_caps: See enum virtchnl2_cap_txrx_csum
+ * @seg_caps: See enum virtchnl2_cap_seg
+ * @hsplit_caps: See enum virtchnl2_cap_rx_hsplit_at
+ * @rsc_caps: See enum virtchnl2_cap_rsc
+ * @rss_caps: See enum virtchnl2_cap_rss
+ * @other_caps: See enum virtchnl2_cap_other
+ * @max_tx_hdr_size: Max header length hardware can parse/checksum, in bytes.
+ * @max_sg_bufs_per_tx_pkt: Max number of scatter gather buffers that can be
+ *			    sent per transmit packet without needing to be
+ *			    linearized.
+ * @min_sso_packet_len: Min packet length supported by device for single
+ *	segment offload
+ * @q_info: Max queue information
+ * @crc_enable: Enable CRC insertion offload
+ */
+struct idpf_eth_idc_auxiliary_dev_caps {
+	__le32 csum_caps;
+	__le32 seg_caps;
+	__le32 hsplit_caps;
+	__le32 rsc_caps;
+	__le64 rss_caps;
+	__le64 other_caps;
+	__le16 max_tx_hdr_size;
+	u8 max_sg_bufs_per_tx_pkt;
+	u8 min_sso_packet_len;
+	struct idpf_max_q q_info;
+	bool crc_enable;
+};
+
+/**
+ * struct idpf_eth_idc_event - ethernet idc events
+ * @event_data: Event data
+ * @event_code: Event code
+ */
+struct idpf_eth_idc_event {
+	void *event_data;
+	enum idpf_eth_idc_event_code event_code;
+};
+
+struct idpf_eth_idc_dev_info;
+/**
+ * struct idpf_eth_idc_ops - Ethernet driver ops to main driver
+ * @event_send: Sends ethernet event to main driver
+ * @virtchnl_send: Sends Virtchnl messages call to main driver
+ * @intr_reg_init: Initialize interrupt registers call to main driver
+ * @intr_init_vec_idx: Initialize the vector indexes call to main driver
+ * @req_rel_vec_idx: Release vector indexes call to main driver
+ */
+struct idpf_eth_idc_ops {
+	void (*event_send)(struct idpf_eth_idc_dev_info *dev_info,
+			   struct idpf_eth_idc_event *event);
+	size_t (*virtchnl_send)(struct idpf_eth_idc_dev_info *dev_info,
+				struct idpf_vc_xn_params *params);
+	int (*intr_reg_init)(struct idpf_eth_idc_dev_info *dev_info,
+			     u16 num_vecs, struct idpf_q_vector *q_vectors,
+			      u16 *q_vector_idxs);
+	int (*intr_init_vec_idx)(struct idpf_eth_idc_dev_info *dev_info,
```

Comment:
This would be direct function call as per past comments during review. Atleast thats how  I have it in OOT

Reply:
I think we have settled all this. All the PCI calls are direct calls from IDC, these are auxiliary calls only function pointer from IDC to aux etc.